src/lib/statistics/chunk_statistics/histograms/equal_width_histogram.hpp

```diff
@@ -0,0 +1,182 @@
+   * This is a theoretical problem for floating point numbers as well, but not solved,
+   * because the number of representable floating point numbers in a range is not trivial to determine.
+   */
+  BinID bin_count() const override;
+
+ protected:
+  BinID _bin_for_value(const T& value) const override;
+  BinID _next_bin_for_value(const T& value) const override;
+
+  T _bin_minimum(const BinID index) const override;
+  T _bin_maximum(const BinID index) const override;
+  HistogramCountType _bin_height(const BinID index) const override;
+  HistogramCountType _bin_distinct_count(const BinID index) const override;
+
+  /**
+   * Creates bins and their statistics.
+   *
+   * This method is implemented in the header file because of std::enable_if_t.
+   * It is only enabled for numerical types.
+   */
+  template <typename Q = T>
+  static std::enable_if_t<std::is_arithmetic_v<Q>, EqualWidthBinData<T>> _build_bins(
+      const std::vector<std::pair<T, HistogramCountType>>& value_counts, const BinID max_bin_count) {
+    // Bins shall have the same range.
+    const auto min = value_counts.front().first;
+    const auto max = value_counts.back().first;
+    const auto base_width = next_value(max - min);
+
+    // Never have more bins than representable values.
+    // TODO(anyone): fix for floats. This is more of a theoretical issue, however.
+    auto bin_count = max_bin_count;
+    if constexpr (std::is_integral_v<T>) {
+      bin_count = max_bin_count <= static_cast<BinID>(base_width) ? max_bin_count : base_width;
+    }
+
+    const auto bin_width = base_width / bin_count;
+    BinID bin_count_with_larger_range;
+    if constexpr (std::is_integral_v<T>) {
+      bin_count_with_larger_range = base_width % bin_count;
+    } else {
+      bin_count_with_larger_range = 0ul;
+    }
+
+    std::vector<HistogramCountType> bin_counts(bin_count);
+    std::vector<HistogramCountType> bin_distinct_counts(bin_count);
+
+    auto current_bin_begin_value = min;
+    auto current_bin_begin_it = value_counts.cbegin();
+    for (auto current_bin_id = BinID{0}; current_bin_id < bin_count; current_bin_id++) {
+      T next_bin_begin_value = current_bin_begin_value + bin_width;
+
+      if constexpr (std::is_integral_v<T>) {
+        if (current_bin_id < bin_count_with_larger_range) {
+          next_bin_begin_value++;
+        }
+      }
+
+      if constexpr (std::is_floating_point_v<T>) {
+        // This is intended to compensate for the fact that floating point arithmetic is not exact.
+        // Adding up floating point numbers adds an error over time.
+        // So this is how we make sure that the last bin contains the rest of the values.
+        if (current_bin_id == bin_count - 1) {
+          next_bin_begin_value = next_value(max);
+        }
+      }
+
+      auto next_bin_begin_it = current_bin_begin_it;
+      while (next_bin_begin_it != value_counts.cend() && (*next_bin_begin_it).first < next_bin_begin_value) {
```

Comment:
you don't need to fix this, just for my understanding: one could do a binary search here, amirite?

Reply:
Yes, I talked to @Bouncner about this at some point and he pointed out that for smaller vectors a linear search is as fast or faster because of hardware optimizations and stuff.
We kept it as is because the number of bins is expected to be small enough for a linear search not to hurt much, if at all.
If it shows up in a profiler I am happy to change it ;)

Reply:
Done.